Studies/python/BayesianOptimizationCustom.py

```diff
@@ -52,19 +60,29 @@ def TransformParams(self, params):
             else:
                 raise Exception('type {} not allowed'.format(self.params_typed_range[key]['type']))
 
+
         for key, value in params.items():
-            #Has some dependency wiht other variables
-            if 'depends_on' in self.params_typed_range[key]:
-                # the variable on which it depends has a value of zero
-                if new_params[self.params_typed_range[key]['depends_on'] ] == 0:
-                    new_params[key] = self.params_typed_range[key]['range'][0]
+            if type(self.params_typed_range[key]['range']) == list:
```

Comment:
I don't understand the purpose of this `if`. You iterate over `params` which are in opt space, where only non-const params should be present. If it is not the case, there are some error elsewhere.